tools/dds/dds-server/dds-server.cpp

```diff
@@ -23,134 +23,202 @@ dds_server::dds_server()
+    _ctx.set_devices_changed_callback( [this]( rs2::event_information & info ) {
+        if( _running )
+        {
+            std::vector< std::string > devices_to_remove;
+            std::vector< std::pair< std::string, rs2::device > > devices_to_add;
+
+            if( prepare_devices_changed_lists( info, devices_to_remove, devices_to_add ) )
+            {
+                // Post the devices connected / removed
+                _dds_device_dispatcher.invoke(
+                    [this, devices_to_remove, devices_to_add]( dispatcher::cancellable_timer c ) {
+                        post_device_changes( devices_to_remove, devices_to_add );
+                    } );
+            }
+        }
+    } );
+
+    std::cout << "RS DDS Server is on.." << std::endl;
+}
+
+bool dds_server::prepare_devices_changed_lists(
+    const rs2::event_information & info,
+    std::vector< std::string > &devices_to_remove,
+    std::vector< std::pair< std::string, rs2::device > > &devices_to_add )
+{
+    // Remove disconnected devices from devices list
+    for( auto dev_info : _devices_writers )
+    {
+        auto & dev = dev_info.second.device;
+        auto device_name = dev.get_info( RS2_CAMERA_INFO_NAME );
+
+        if( info.was_removed( dev ) )
+        {
+            devices_to_remove.push_back( device_name );
+        }
+    }
+
+    // Add new connected devices from devices list
+    for( auto && dev : info.get_new_devices() )
+    {
+        auto device_name = dev.get_info( RS2_CAMERA_INFO_NAME );
+        devices_to_add.push_back( { device_name, dev } );
+    }
+
+    bool device_change_detected = !devices_to_remove.empty() || !devices_to_add.empty();
+    return device_change_detected;
+}
+
+void dds_server::post_device_changes(
+    std::vector< std::string > devices_to_remove,
+    std::vector< std::pair< std::string, rs2::device > > devices_to_add )
+{
+    try
+    {
+        for( auto dev_to_remove : devices_to_remove )
+        {
+            remove_dds_device( dev_to_remove );
+        }
+
+        for( auto dev_to_add : devices_to_add )
+        {
+            add_dds_device( dev_to_add );
+        }
+    }
+
+    catch( ... )
+    {
+        std::cout << "Unknown error when trying to remove/add a DDS device" << std::endl;
+    }
+}
+
+void dds_server::remove_dds_device( std::string device_name )
+{
+    // deleting a device also notify the clients internally
+    _publisher->delete_datawriter( _devices_writers[device_name].data_writer );
+    std::cout << "Device '" << device_name << "' - removed" << std::endl;
+    _devices_writers.erase( device_name );
+}
+
+void dds_server::add_dds_device( std::pair< std::string, rs2::device > device_pair )
+{
+    auto dev_name = device_pair.first;
+    auto rs2_dev = device_pair.second;
+
+    if( !create_device_writer( dev_name, rs2_dev) )
+    {
+        std::cout << "Error creating a DDS writer" << std::endl;
```

Comment:
Need to standardize the way we output information/errors.
In general, I want to avoid stdout and assume this will be a service, meaning use logs.
Maybe in next PR?